x/gamm/pool-models/balancer/amm.go

```diff
@@ -250,16 +250,25 @@ func (p *Pool) calcSingleAssetJoin(tokenIn sdk.Coin, swapFee sdk.Dec, tokenInPoo
 	).TruncateInt(), nil
 }
 
+// JoinPool calculates the number of shares needed and the updated liquidity for joining pool
```

Comment:
Also, consider elaborating on what kind of errors are returned and given which inputs / state of the pool

```suggestion
// JoinPool calculates the number of shares needed given tokensIn with swapFee applied.
// It updates the liquidity if the pool is joined successfully. If not, returns error.
```